src/Regulator.php

```diff
@@ -349,7 +353,8 @@ public function process_task( string $args_hash ): void {
 			$task = Tasks_Table::get_by_args_hash( $args_hash );
 
 			if ( ! $task ) {
-				throw new RuntimeException( 'No Shepherd task found with args hash ' . $args_hash . '.' );
+				// translators: %s is the arguments hash.
+				throw new RuntimeException( sprintf( esc_html__( 'No Shepherd task found with args hash %s.', 'stellarwp-shepherd' ), $args_hash ) );
```

Comment:
Here and elsewhere, use `esc_html_x` and provide a translators note for the placeholders:
```suggestion
				throw new RuntimeException( sprintf(
				    // translators: the %s is the arguments hash.
				    esc_html_x( 'No Shepherd task found with args hash %s.', 'User facing error message', 'stellarwp-shepherd' ), $args_hash ) );
```